src/plugins/kibana/public/settings/sections/indices/add_data_steps/pattern_review_step.js

```diff
@@ -9,6 +69,78 @@ modules.get('apps/settings')
         sampleDocs: '=',
         indexPattern: '=',
         pipeline: '='
+      },
+      controllerAs: 'reviewStep',
+      bindToController: true,
+      controller: function ($scope, Private) {
+        this.sampleDocs = testData;
+        this.pipeline = testPipeline;
+
+        if (_.isUndefined(this.indexPattern)) {
+          this.indexPattern = {};
+        }
+
+        const knownFieldTypes = {};
+        this.dateFields = [];
+        this.pipeline.forEach((processor) => {
+          if (processor.geoip) {
+            const field = processor.geoip.target_field || 'geoip';
+            knownFieldTypes[field] = 'geo_point';
+          }
+          else if (processor.date) {
+            const field = processor.date.target_field || '@timestamp';
```

Comment:
The ES ingest API simply overwrites the target_field if it already exists, so I don't think we have to worry about conflicts.